### PR TITLE
Create a `Migrating from Log4j 1 to Log4j 2` guide

### LINKED ISSUE #2519: Create a `Migrating from Log4j 1 to Log4j 2` guide
As agreed in [the site structure document](https://docs.google.com/document/d/10Fu7oqDzdM_D6LbexzwX9arh51Tic7AGvkWTQrL6jjQ/edit), create a _"Migrating from Log4j 1 to Log4j 2"_ guide such that

- Move the content from `manual/migrate.adoc`
- Target `2.x`, port to `main`

### PR DESCRIPTION
**Pull request by ppkarwasz**
Closes #2519.

We restructure the pages related to migration between Log4j 1 and Log4j 2 to provide a new "Step by step" guide on migrating between the two major version.

The new guide formally introduces the concept of "Log4j 1 API" and provides the user two upgrade paths:

1. If the application or one of its libraries uses Log4j 1 API directly (very unlikely, the usage of Log4j 1 as logging API has declined since the introduction of JCL 22 years ago) the user must apply a set of conversion rules to its source code. We also add a reference to [an OpenRewrite rule](https://docs.openrewrite.org/recipes/java/logging/log4j/log4j1tolog4j2) that does it automatically.
2. If the application uses Log4j 1 as a backend, users can choose a shorter upgrade path that only requires a change in the runtime dependencies. I plan to contribute a separate OpenRewrite recipe for this in #2080 (cf. openrewrite/rewrite-logging-frameworks#154).
3. Conversion of configuration files is treated separately, since there are applications that don't use Log4j 1 any more (neither as logging API or logging backend), but need to provide backward compatibility for their users.

The section about "Log4j 1 to Log4j 2 bridge" is splitted among the 3 sections above:

1. Log4j 1 API users only need to add the bridge if they can not modify their source code (or upgrade libraries),
2. Log4j 1 backend users only need to add the bridge if they need to use native Log4j 1 components (and there is no Log4j 2 component that offers the same features),
3. The "Log4j 1 to Log4j 2 bridge" is also required in the rare case, when application developers want to provide BC of configuration files to their users. IIRC Kafka, Hadoop or Spark use this feature.

### CODE REVIEW

File: src/site/antora/modules/ROOT/examples/manual/migration/Migration1Example.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package example;
+
+import org.apache.log4j.Logger;
+
+public class Migration1Example {
+
+    private static final Logger logger = Logger.getLogger(Migration1Example.class);
```
Comment (vy) [inline src/site/antora/modules/ROOT/examples/manual/migration/Migration1Example.java:23]: Could you rename all `static final`s to upper-case, please? That is, `s/logger/LOGGER/`. Note that this is the recommended practice in `getting-started.adoc` too.

File: src/site/antora/modules/ROOT/partials/manual/systemproperties/properties-log4j-12-api.adoc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+|
+https://docs.oracle.com/javase/{java-target-version}/docs/api/java/nio/file/Path.html[Path]
+or
+https://docs.oracle.com/javase/{java-target-version}/docs/api/java/net/URI.html[URI]
+
+| Default value
+| `null`
+|===
+
+If not `null`, Log4j will try to retrieve a **Log4j 1** configuration file from the given location.
```
Comment (vy) [inline src/site/antora/modules/ROOT/partials/manual/systemproperties/properties-log4j-12-api.adoc:52]: Increasing the emphasis around 1-vs-2 as was done above.

```suggestion
If not `null`, Log4j 2 will try to retrieve a **Log4j 1** configuration file from the given location.
```

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc [hunk 3]
```diff
@@ -16,433 +16,573 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
-</log4j:configuration>
-----
+| {log4j1-url}/apidocs/org/apache/log4j/ConsoleAppender.html[org.apache.log4j.ConsoleAppender]
+| xref:manual/appenders.adoc#ConsoleAppender[Console]
+|
 
-Log4j 2 XML configuration
+| {log4j1-url}/apidocs/org/apache/log4j/DailyRollingFileAppender.html[org.apache.log4j.DailyRollingFileAppender]
+| xref:manual/appenders.adoc#RollingFileAppender[RollingFile]
+| See <<rolling-file-conversion>>
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/migration.adoc:363]: Link is missing a label: 
image

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc [hunk 4]
```diff
@@ -16,433 +16,573 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
-</Configuration>
-----
+| {log4j1-url}/apidocs/org/apache/log4j/FileAppender.html[org.apache.log4j.FileAppender]
+| xref:manual/appenders.adoc#FileAppender[File]
+|
 
-=== Sample 2 - Migrating a simple File Appender, XMLLayout and SimpleLayout configuration
+| {log4j1-url}/apidocs/org/apache/log4j/RollingFileAppender.html[org.apache.log4j.RollingFileAppender]
+| xref:manual/appenders.adoc#RollingFileAppender[RollingFile]
+| See <<rolling-file-conversion>>
```
Comment (vy) [inline src/site/antora/modules/ROOT/pages/manual/migration.adoc:371]: Link is missing a label:
image

**Review by vy — CHANGES_REQUESTED** [review]

File: src/site/antora/modules/ROOT/examples/manual/migration/Migration1Example.java (hunk 1 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/examples/manual/migration/Migration1Example.java]: Fixed in 82a6f08f9aadc6043d5594fac9bf7589912c676b

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc (hunk 3 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/migration.adoc]: Fixed in 4b38c891c6a268e27ad8102e2a49004a4877524e

File: src/site/antora/modules/ROOT/pages/manual/migration.adoc (hunk 4 above)
Comment (ppkarwasz) [reply src/site/antora/modules/ROOT/pages/manual/migration.adoc]: Fixed in 4b38c891c6a268e27ad8102e2a49004a4877524e

**Pull request merged**